doc/vxlan/EVPN/EVPN_VxLAN_Multihoming.md

```diff
@@ -0,0 +1,1450 @@
+| ---- | --------- | ------------------------------------------------ | ------------------ |
+| 0.1  | 2/22/2024 | Syed Hasan Naqvi, Rajesh Sankaran, Kishore Kunal, Praveen Elagala | Initial version    |
+
+# About this Manual
+
+This document provides general information of EVPN VxLAN Multihoming feature implementation in SONiC.
+
+
+
+# Scope
+
+This document describes the high level design and changes required for supporting Multihoming in EVPN VxLAN networks.
+This feature will provide RFC 7432/8365 based Multihoming solution in SONiC.
+The ICCPd based MCLAG will continue to co-exist as an alternate solution.
+
+
+
+
+# Definition/Abbreviation
+
+## Table 1: Abbreviations
+
+| Term     | Meaning                                   |
+| -------- | ----------------------------------------- |
+| MCLAG    | Multi-Chassis LAG                         |
+| LVTEP    | Logical VTEP                              |
+| PIP, VIP | Primary IP Address, Virtual IP Address    |
+| MHD, SHD | Multi-Homed Device, Single-Homed Device   |
+| VNI      | (VxLAN) Virtual Network Identifier        |
+| L3VNI    | (VxLAN) Layer3 Virtual Network Identifier |
+| IRB VLAN | Integrated Routing and Bridging VLAN      |
+
+# 1 Feature Overview
+
+
+
+
+
+## 1.1 Requirements
+
+### 1.1.1 Functional Requirements
+
+The following are functional requirements for EVPN VxLAN Multihoming:
+
+1. Support All-EVPN based active-active access redundancy with up to 4 VTEPs.
+2. Support all of the routing and switching traffic flows currently supported on legacy MCLAG.
+3. Support following features on EVPN MH:
+
+	- Static Anycast Gateway
+	- BGP over EVPN Multihomed Ethernet-Segment
+	- ARP/ND suppression
+4. Support for Type-0 (Operator configured), Type-1 (LACP based), and Type-3 (System-MAC config based) Ethernet Segment IDs types.
+
+The following are non-goals for the EVPN Multihoming feature:
+
+1. Formation of EVPN Active-Active redundancy with multiple vendor NOS. All of the VTEPs, which the hosts are Multihomed to, should be SONiC devices.
+2. Single-Active redundancy for the multihomed device. Only active-active redundancy is supported.
+3. Support of VRRP over EVPN Multihoming. Static Anycast Gateway should be used instead.
+4. EVPN Multihomed interface as router-port or routed sub-interface. EVPN Multihomed interface can only be a switchport.
+
+Special notes:
+1. SONiC currently supports EVPN Vlan-based service. No new service-types are introduced with this feature. 
+
+
+### 1.1.2 Configuration and Management Requirements
+
+The following are configuration and management requirements for EVPN VxLAN Multihoming:
+
+1. All of the configurations and show outputs should be available through SONiC yang - REST, gNMI.
+2. All of the configurations and show outputs should be available through click/FRR-vtysh.
+
+### 1.1.3 Platform requirements
+Support EVPN Multihoming on platforms having VxLAN capabilities.
+
+### 1.1.4 Scalability Requirements
```

Comment:
The solution is vendor agnostic, it's difficult to claim scaling performance and convergence in isolation of platform details and ASIC capabilities, please remove section 1.1.4, 1.1.5, and section 8

Reply:
Agreed. Removed.